full/src/main/java/apoc/systemdb/metadata/ExportMetadata.java

```diff
@@ -0,0 +1,57 @@
+package apoc.systemdb.metadata;
+
+import apoc.SystemLabels;
+import apoc.SystemPropertyKeys;
+import org.apache.commons.lang3.StringUtils;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.internal.helpers.collection.Pair;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public interface ExportMetadata {
+    
+    enum Type {
+        CypherProcedure(new ExportProcedure()),
+        CypherFunction(new ExportFunction()),
+        Uuid(new ExportUuid()),
+        Trigger(new ExportTrigger()),
+        DataVirtualizationCatalog(new ExportDataVirtualization());
+
+        private final ExportMetadata exportMetadata;
+
+        Type(ExportMetadata exportMetadata) {
+            this.exportMetadata = exportMetadata;
+        }
+        
+        public Stream<Pair<String, String>> export(Node node) {
+            return exportMetadata.export(node);
+        }
+        
+        public static Optional<Type> from(Label label) {
+            final String name = label.name();
+            if (name.equalsIgnoreCase(SystemLabels.Procedure.name())) {
+                return Optional.of(CypherProcedure);
+            } else if(name.equalsIgnoreCase(SystemLabels.Function.name())) {
+                return Optional.of(CypherFunction);
+            } else if(name.equalsIgnoreCase(SystemLabels.ApocTrigger.name())) {
+                return Optional.of(Trigger);
+            } else if(name.equalsIgnoreCase(SystemLabels.ApocUuid.name())) {
+                return Optional.of(Uuid);
+            } else if(name.equalsIgnoreCase(SystemLabels.DataVirtualizationCatalog.name())) {
+                return Optional.of(DataVirtualizationCatalog);
+            }
+            return Optional.empty();
+        }
+    }
+
+    Stream<Pair<String, String>> export(Node node);
```

Comment:
ti avevo chiesto di metterlo così
```suggestion
    List<Pair<String, String>> export(Node node);
```

Reply:
Scusa non ho capito in che modo preciso vuoi cambiarlo, cioè se lo metto così non posso fare la `groupingBy` in `SystemDb.java` come sta adesso